okhttp/src/main/java/okhttp3/Authenticator.java

```diff
@@ -62,11 +70,21 @@
 public interface Authenticator {
   /** An authenticator that knows no credentials and makes no attempt to authenticate. */
   Authenticator NONE = new Authenticator() {
+    @Override public Request authenticate(Request request) {
+      return null;
+    }
+
     @Override public Request authenticate(Route route, Response response) {
       return null;
     }
   };
 
+  /**
+   * Returns a tunnel request if a TLS tunnel via an HTTP proxy. Returns a request before
```

Comment:
I don't really follow the docs here.  Is this just saying it can be used either for proxy authentication or request authentication?